### ISSUE TITLE
Expand docs for MA0099

### ISSUE THREAD

**Issue by erri120** [meziantou/Meziantou.Analyzer#525]
Just upgraded to 2.0.49, and I'm now getting [MA0099](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0099.md) warnings:

https://github.com/erri120/GameFinder/blob/7081bfa6909722c33099ac14c2403de7d0b6064d/src/GameFinder.StoreHandlers.Steam/SteamGameId.cs#L8-L9

```
warning MA0099: Use Explicit enum value for 'Customizations' instead of 0
warning MA0099: Use Explicit enum value for 'DebuggerAttributeGeneration' instead of 0
```

I'm guessing the warning comes from the use of some default values:

https://github.com/SteveDunn/Vogen/blob/415360f0ce41b4bdc2e80ddb0a6da5328b9dcb1e/src/Vogen.SharedTypes/ValueObjectAttribute.cs#L17-L25

Is MA0099 supposed to produce warnings for default values? Could you expand the docs for MA0099 and explain what triggers this warning, why it's considered a warning, and how you can fix this?

**Comment by meziantou** [comment]
Default values should be handled correctly. It looks like a bug, I'll have a look at it soon.

**Comment by meziantou** [comment]
I cloned your project, but I do not get any MA0099. Just to be sure I added `dotnet_diagnostic.MA0099.severity = error` and run `dotnet build`.

Could you try to create a simple repro?

**Comment by meziantou** [comment]
Which version of Roslyn and which language version do you use?
You can use `#error version` to get the information. See [Getting Roslyn (C# compiler) and Language versions](https://www.meziantou.net/getting-roslyn-csharp-compiler-and-language-versions.htm) for more info.

**Comment by erri120** [comment]
> Which version of Roslyn and which language version do you use? You can use `#error version` to get the information. See [Getting Roslyn (C# compiler) and Language versions](https://www.meziantou.net/getting-roslyn-csharp-compiler-and-language-versions.htm) for more info.

```
Compiler version: '4.4.0-6.23068.5 (0760f14e)'. Language version: 11.0.
```

**Comment by meziantou** [comment]
I can repro on Roslyn 4.4 and before, but it's fine on 4.5 (latest).
I will investigate later. As a workaround, you can update VS or the .NET SDK.
